src/overlays/actors/ovl_En_River_Sound/z_en_river_sound.c

```diff
@@ -27,15 +26,220 @@ const ActorInit En_River_Sound_InitVars = {
     (ActorFunc)EnRiverSound_Update,
     (ActorFunc)EnRiverSound_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_River_Sound/EnRiverSound_Destroy.s")
+void EnRiverSound_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnRiverSound* this = THIS;
+
+    this->unk_14C = 0;
+    this->pathIndex = (this->actor.params >> 8) & 0xFF;
+    this->actor.params = this->actor.params & 0xFF;
+
+    if (this->actor.params >= 248) {
+        func_800F4870(this->actor.params - 248);
+        Actor_Kill(&this->actor);
+    } else if (this->actor.params == 247) {
+        func_800F6FB4(4);
+        Actor_Kill(&this->actor);
+    } else if (this->actor.params == 12) {
+        u32 q = gSaveContext.inventory.questItems;
+        if (!(gBitFlags[12] & q) || (gBitFlags[14] & q))
+            Actor_Kill(&this->actor);
+    }
+}
+
+void EnRiverSound_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnRiverSound* this = THIS;
+
+    if (this->actor.params == 12)
```

Comment:
We require braces for single line blocks based on our current code style. If you run clang-tidy (or just `format.sh`), it should fix these for you.

Reply:
Done.